src/hotspot/share/jfr/leakprofiler/utilities/unifiedOopRef.hpp

```diff
@@ -29,21 +29,33 @@
 #include "utilities/globalDefinitions.hpp"
 
 struct UnifiedOopRef {
+  static const uintptr_t tag_mask   = LP64_ONLY(0b111) NOT_LP64(0b011);
+  static const uintptr_t native_tag = 0b001;
+  static const uintptr_t raw_tag    = 0b010;
```

Comment:
What is "raw", compared to "native"?

Reply:
Native is native storage (not in java heap) which requires GC barriers. Raw is raw storage which does not require GC barriers. This is thread local storage, such as the handle area, stack, etc.